### PR TITLE
Simplify rule according to feedback in #531

### PR DESCRIPTION
**Pull request by simsurace**
There were some suggestions in #531 after it was closed. This PR implements them.

### CODE REVIEW

File: src/chainrules.jl [hunk 1]
```diff
@@ -111,10 +111,9 @@ end
 
 function ChainRulesCore.rrule(s::Sinus, x::AbstractVector, y::AbstractVector)
     d = x - y
-    sind = sinpi.(d)
-    abs2_sind_r = abs2.(sind) ./ s.r .^ 2
+    abs2_sind_r = (sinpi.(d) ./ s.r) .^ 2
     val = sum(abs2_sind_r)
-    gradx = twoπ .* cospi.(d) .* sind ./ s.r .^ 2
+    gradx = π .* cospi.(2 .* d) ./ s.r .^ 2
```
Comment (devmotion) [inline src/chainrules.jl:116]: I misremembered the formula 🙈 
```suggestion
    gradx = π .* sinpi.(2 .* d) ./ s.r .^ 2
```

File: src/chainrules.jl [hunk 2]
```diff
@@ -136,7 +135,7 @@ function ChainRulesCore.rrule(
             for j in 1:n, i in 1:n
                 xi = view(x, i, :)
                 xj = view(x, j, :)
-                ds = twoπ .* Δ[i, j] .* sinpi.(xi .- xj) .* cospi.(xi .- xj) ./ d.r .^ 2
+                ds = π .* Δ[i, j] .* cospi.(2 .* (xi .- xj)) ./ d.r .^ 2
```
Comment (devmotion) [inline src/chainrules.jl:138]:
```suggestion
                ds = π .* Δ[i, j] .* sinpi.(2 .* (xi .- xj)) ./ d.r .^ 2
```

File: src/chainrules.jl [hunk 3]
```diff
@@ -173,19 +172,19 @@ function ChainRulesCore.rrule(
             for j in 1:m, i in 1:n
                 xi = view(x, i, :)
                 yj = view(y, j, :)
-                ds = twoπ .* Δ[i, j] .* sinpi.(xi .- yj) .* cospi.(xi .- yj) ./ d.r .^ 2
+                ds = π .* Δ[i, j] .* cospi.(2 .* (xi .- yj)) ./ d.r .^ 2
```
Comment (devmotion) [inline src/chainrules.jl:175]:
```suggestion
                ds = π .* Δ[i, j] .* sinpi.(2 .* (xi .- yj)) ./ d.r .^ 2
```

File: src/chainrules.jl [hunk 4]
```diff
@@ -173,19 +172,19 @@ function ChainRulesCore.rrule(
...
-                ȳ[j, :] -= ds
+                x̄[i, :] .+= ds
+                ȳ[j, :] .-= ds
             end
         elseif dims == 2
             for j in 1:m, i in 1:n
                 xi = view(x, :, i)
                 yj = view(y, :, j)
-                ds = twoπ .* Δ[i, j] .* sinpi.(xi .- yj) .* cospi.(xi .- yj) ./ d.r .^ 2
+                ds = π .* Δ[i, j] .* cospi.(2 .* (xi .- yj)) ./ d.r .^ 2
```
Comment (devmotion) [inline src/chainrules.jl:184]:
```suggestion
                ds = π .* Δ[i, j] .* sinpi.(2 .* (xi .- yj)) ./ d.r .^ 2
```

File: src/chainrules.jl [hunk 5]
```diff
@@ -208,10 +207,10 @@ function ChainRulesCore.rrule(
         for i in 1:n
             xi = view(x, :, i)
             yi = view(y, :, i)
-            ds = twoπ .* Δ[i] .* sinpi.(xi .- yi) .* cospi.(xi .- yi) ./ d.r .^ 2
+            ds = π .* Δ[i] .* cospi.(2 .* (xi .- yi)) ./ d.r .^ 2
```
Comment (devmotion) [inline src/chainrules.jl:210]:
```suggestion
            ds = π .* Δ[i] .* sinpi.(2 .* (xi .- yi)) ./ d.r .^ 2
```

File: src/chainrules.jl (hunk 1 above)
Comment (simsurace) [reply src/chainrules.jl:116]: No worries!

**Pull request merged**